src/cargo/ops/cargo_new.rs

```diff
@@ -759,69 +759,53 @@ fn mk(config: &Config, opts: &MkOptions<'_>) -> CargoResult<()> {
     init_vcs(path, vcs, config)?;
     write_ignore_file(path, &ignore, vcs)?;
 
-    let mut cargotoml_path_specifier = String::new();
+    // Create `Cargo.toml` file with necessary `[lib]` and `[[bin]]` sections, if needed.
+    let mut manifest = toml_edit::Document::new();
+    manifest["package"] = toml_edit::Item::Table(toml_edit::Table::new());
+    manifest["package"]["name"] = toml_edit::value(name);
+    manifest["package"]["version"] = toml_edit::value("0.1.0");
+    let edition = match opts.edition {
+        Some(edition) => edition.to_string(),
+        None => Edition::LATEST_STABLE.to_string(),
+    };
+    manifest["package"]["edition"] = toml_edit::value(edition);
+    if let Some(registry) = opts.registry {
+        let mut array = toml_edit::Array::default();
+        array.push(registry);
+        array.decor_mut().set_suffix("\n\n# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html");
+        manifest["package"]["publish"] = toml_edit::value(array);
+    } else {
+        manifest["package"]["edition"].as_value_mut().expect("edition is a string value").decor_mut().set_suffix("\n\n# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html");
+    }
+    manifest["dependencies"] = toml_edit::Item::Table(toml_edit::Table::new());
 
     // Calculate what `[lib]` and `[[bin]]`s we need to append to `Cargo.toml`.
-
     for i in &opts.source_files {
         if i.bin {
             if i.relative_path != "src/main.rs" {
-                cargotoml_path_specifier.push_str(&format!(
-                    r#"
-[[bin]]
-name = "{}"
-path = {}
-"#,
-                    i.target_name,
-                    toml::Value::String(i.relative_path.clone())
-                ));
+                let mut bin = toml_edit::Table::new();
+                bin["name"] = toml_edit::value(i.target_name.clone());
+                bin["path"] = toml_edit::value(i.relative_path.clone());
+                manifest["bin"]
+                    .or_insert(toml_edit::Item::ArrayOfTables(
```

Comment:
Why `or_insert`?  you know there is no `bin` key

Reply:
Because this is a for loop.

Reply:
Right, the downside of looking too closely at a change and not the context.